### ISSUE TITLE
Stable/atlantis  Issue Regarding StatefulSet

### ISSUE THREAD

**Issue by justinhauer** [helm/charts#12530]
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

This is a bug report.

When using image pull secrets outlined in the helm chart:

https://github.com/helm/charts/blob/352f6a4c7a442b2ef65c34fb549602f0b0dce8f7/stable/atlantis/templates/statefulset.yaml#L232

The current location where imagepullSecrets is located in the stateful set causes this to incorrectly appear within the container spec of the manifest, causing the Helm deploy to not successfully run:
```
           failureThreshold: 5
          volumeMounts:
          - name: atlantis-data
            mountPath: /atlantis-data
          resources:
      imagePullSecrets:
      - name: mypullsecrethere
            limits:
              cpu: 100m
              memory: 1Gi
            requests:
              cpu: 100m
              memory: 1Gi
```
Moving the if block:
```
      {{- if .Values.imagePullSecrets }}
      imagePullSecrets:
        - name: {{ .Values.imagePullSecrets }}
      {{- end }}
```
Further down in the code resolves this issue. Please resolve.

**Version of Helm and Kubernetes**:
helm: versions client[v2.13.0] server[v2.12.1]

kubernetes: 1.9.6

**Which chart**:

https://github.com/helm/charts/blob/352f6a4c7a442b2ef65c34fb549602f0b0dce8f7/stable/atlantis/templates/statefulset.yaml#L232

**What happened**:
Helm Chart will not deploy, malformed yaml when using imagePullSecret to pull image from private repo

**What you expected to happen**:

the helm chart will deploy
**How to reproduce it** (as minimally and precisely as possible):
Re-running the helm install. Resolved by editing the statefulset yaml

**Anything else we need to know**:
@switchboardOp had placed in a PR to resolve the issue

**Comment by anoora17** [comment]
I have exactly same problem:
helm version "v2.12.3" and tiller  "v2.13.0"
Kubernetes is EKS kubernetes version 1.11

**Comment by lkysow** [comment]
Fixed by #12506
